### ISSUE TITLE
Add conf-bash for BSDs

### ISSUE THREAD

**Issue by kit-ty-kate** [ocaml/opam-repository#22457]
zarith and gmp could be easy target to add a dependency for them
```
File "duniverse/Zarith/dune", line 79, characters 0-146:
79 | (rule
80 |  (deps META)
81 |  (action
82 |   (with-stdout-to
83 |    zarith_version.ml
84 |    (progn
85 |     (run echo "let")
86 |     (bash "grep \"version\" META | head -1")))))
Error: I need bash to interpret (bash ...) actions but I couldn't find it :(
```

**Comment by raphael-proust** [comment]
https://github.com/ocaml/opam-repository/pull/22516 ← This pull request introduced some packages that could use a build-time dependency to `conf-bash`

**Comment by jideelgh** [comment]
I'm using OpenBSD 7.4 with ksh, nothing fancy.
conf-bash keeps asking to install bash every time i launch "opam upgrade" while it's already installed.
I tried to let opam install bash or to install it manually.
Bash is installed in /usr/local/bin/bash, I tried to symlink to /bin/bash, no success.
bash --version returns 
"GNU bash, version 5.2.15(1)-release (x86_64-unknown-openbsd7.4)
Copyright (C) 2022 Free Software Foundation, Inc.
[...]"
I used ktrace to investigate some pledge/unveil issue, couldn't identify one.
I also tried to start opam w/ the --debug flag & adjusting the level with --debug-level flags, no luck either.
Any idea ?
